ratelimiter.php

```diff
@@ -27,26 +27,47 @@
 class RateExceededException extends Exception {}
 
 class RateLimiter {
-	private $prefix, $memcache;
+	private $prefix, $memcache , $keysvisited;
+	// how long should we keep memcache entries
+	public $maxMinutes=10;
+
 	public function __construct(Memcache $memcache, $ip, $prefix = "rate") {
 		$this->memcache = $memcache;
+		if (!$memcache) {
+			echo "Problem connecting to memcache server";
```

Comment:
In accordance with the rest of the code, this should not output something but throw an exception.